Tycho 4: ClassNotFoundException: org.eclipse.aether.spi.synccontext.SyncContextFactory

Hi,

I just tried our build with Tycho 4.0.0-SNAPSHOT and it fails with
```
constituent[180]: file:/home/peter/GODYO-P4-master5/eclipse/configuration/org.eclipse.osgi/711/0/.cp/
---------------------------------------------------
Exception in thread "main" java.lang.NoClassDefFoundError: org/eclipse/aether/spi/synccontext/SyncContextFactory
	at org.eclipse.tycho.core.resolver.MavenTargetLocationFactory.resolveTargetDefinitionContent(MavenTargetLocationFactory.java:53)
	at org.eclipse.tycho.p2resolver.TargetDefinitionResolver.resolveContentWithExceptions(TargetDefinitionResolver.java:172)
	at org.eclipse.tycho.p2resolver.TargetDefinitionResolver.resolveContentWithExceptions(TargetDefinitionResolver.java:193)
	at org.eclipse.tycho.p2resolver.TargetDefinitionResolver.resolveContentWithExceptions(TargetDefinitionResolver.java:193)
	at org.eclipse.tycho.p2resolver.TargetDefinitionResolver.resolveContent(TargetDefinitionResolver.java:102)
	at org.eclipse.tycho.p2resolver.TargetDefinitionResolverService.resolveFromArguments(TargetDefinitionResolverService.java:93)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at org.eclipse.tycho.p2resolver.TargetDefinitionResolverService.getTargetDefinitionContent(TargetDefinitionResolverService.java:68)
	at org.eclipse.tycho.p2resolver.TargetPlatformFactoryImpl.resolveTargetDefinitions(TargetPlatformFactoryImpl.java:214)
	at org.eclipse.tycho.p2resolver.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:164)
	at org.eclipse.tycho.p2resolver.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:137)
	at org.eclipse.tycho.p2resolver.TargetPlatformFactoryImpl.createTargetPlatform(TargetPlatformFactoryImpl.java:88)
	at org.eclipse.tycho.p2resolver.ReactorRepositoryManagerImpl.computePreliminaryTargetPlatform(ReactorRepositoryManagerImpl.java:71)
	at org.eclipse.tycho.p2resolver.P2DependencyResolver.computePreliminaryTargetPlatform(P2DependencyResolver.java:205)
	at org.eclipse.tycho.core.resolver.DefaultTychoResolver.resolveProject(DefaultTychoResolver.java:101)
	at org.eclipse.tycho.core.maven.TychoProjectExecutionListener.beforeProjectLifecycleExecution(TychoProjectExecutionListener.java:108)
	at org.apache.maven.lifecycle.internal.CompoundProjectExecutionListener.beforeProjectLifecycleExecution(CompoundProjectExecutionListener.java:52)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:114)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:577)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: java.lang.ClassNotFoundException: org.eclipse.aether.spi.synccontext.SyncContextFactory
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
	... 33 more
```
what could be the reason? What additional information is needed?

Thanks for looking at the issue.

Bye Peter

--- a/RELEASE_NOTES.md
+++ b/RELEASE_NOTES.md
@@ -7,6 +7,10 @@ This page describes the noteworthy improvements provided by each release of Ecli
 
 ## 4.0.0
 
+### Maven 3.9 required
+
+Tycho 4.x requires Maven Version 3.9
+
 ### creating maven p2 sites with Tycho packaging
 
 There is already a way to [create a p2 maven site with Tycho](https://github.com/eclipse-tycho/tycho/blob/master/RELEASE_NOTES.md#create-p2-repository-referencing-maven-artifacts) for plain jar based projects.

